editor/gdextension_reload_node.cpp

```diff
@@ -0,0 +1,79 @@
+/**************************************************************************/
+/*  gdextension_reload_node.cpp                                           */
+/**************************************************************************/
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "gdextension_reload_node.h"
+
+#include "core/extension/gdextension_manager.h"
+#include "editor/editor_inspector.h"
+#include "editor/editor_interface.h"
+
+void GDExtensionReloadNode::_bind_methods() {
+}
+
+void GDExtensionReloadNode::_notification(int p_what) {
+	switch (p_what) {
+		case NOTIFICATION_APPLICATION_FOCUS_OUT: {
+			unfocused = true;
+		} break;
+
+		case NOTIFICATION_APPLICATION_FOCUS_IN: {
+			callable_mp(this, &GDExtensionReloadNode::_deferred_reload).call_deferred();
+		} break;
+	}
+}
+
+void GDExtensionReloadNode::_deferred_reload() {
```

Comment:
Do we actually need an entire node for just his one method? Can it not be moved to `GDExtensionManager` with `EditorNode` handling the notifications and calling into the manager via public methods?

Reply:
Yes, we could add the notification part to `EditorNode`. I'd usually try not to add more stuff there, but if you think that'd be better, I can move it!

Reply:
I generally agree with not adding stuff to `EditorNode` (as I'm actively cleaning it up), but in this case I expect there to be only 2 new lines. `EditorNode` already handles these notifications, and if you put the implementation of the reload completely into the manager, you only need to call its methods. So just two lines where its appropriate, should be much more compact 👍

Reply:
Er, I started doing this, adding a `GDExtensionManager::reload_extensions()` to do the real work, but this would mean having the `EditorInterface` bit in there too, and we can't call out to editor code in `core/` (as I have been scolded for doing that in the past :-))

So, I think we'd have to move the code that does the actual work into `EditorNode` as well. Is this still the preferred option if all the code needs to come over to `EditorNode` rather than just the two lines?

Reply:
That's unfortunate. Can always use a signal?

Reply:
Hm. So, `EditorNode` will detect the focus change and call `GDExtensionManager::reload_extensions()` which will end by emitting a signal, maybe, `extensions_reloaded`. Now something can connect to the signal in order to call `EditorInterface::get_singleton()->get_inspector()->update_tree()`, but what something should do that? Does `EditorNode` connect to the `extensions_reloaded` signal on `GDExtensionManager` so that we've come full circle?

Reply:
It's fine! I ended up adding a new method, because that just seemed better, but I can remove the method and connect directly to `EditorInterface` if that's what you'd prefer. Anyway, take a look at what's in my latest push and let me know if it's OK!